src/persist-client/src/internal/machine.rs

```diff
@@ -268,7 +268,7 @@ where
         }
     }
 
-    pub async fn merge_res(&mut self, res: &FueledMergeRes<T>) -> bool {
+    pub async fn merge_res(&mut self, res: &FueledMergeRes<T>) -> (bool, ApplyMergeResult) {
```

Comment:
couldn't we maintain an ApplyMergeResult instead of the bool, something analogous to apply_merge_result_ever_true? if so, then this is a bit redundant. i'm happy for this to just be left as a TODO, if it does make sense

Reply:
I'll take a look -- I think I initially avoided changing anything here because the logic is sensitive, and I was concerned about the sprawl of the ApplyMergeResult type change reaching too far. lemme revisit now